kernel/arch/dreamcast/sound/arm/main.c

```diff
@@ -28,12 +28,32 @@ void timer_wait(uint32 jiffies) {
 
 #include <stddef.h>
 
-void * memcpy(void *dest, const void *src, size_t count) {
-    unsigned char *tmp = (unsigned char *) dest;
-    unsigned char *s = (unsigned char *) src;
+void *memcpy(void *dest, const void *src, size_t count) {
+    uint8 *dest8 = (uint8 *)dest;
+    const uint8 *src8 = (const uint8 *)src;
+    uint32 *dest32;
+    const uint32 *src32;
+
+    /* If both src and dest are 4-byte aligned */
+    if(((unsigned long)dest & 3) == 0 && ((unsigned long)src & 3) == 0) {
```

Comment:
Is there a reason to use `unsigned long` for these rather than the `uint32` used otherwise?

Reply:
Nope, fixed.